lib/decimal.ex

```diff
@@ -1246,25 +1246,25 @@ defmodule Decimal do
 
   ## Examples
 
-      iex> Decimal.from_any(3)
+      iex> Decimal.cast(3)
       #Decimal<3>
 
-      iex> Decimal.from_any(3.0)
+      iex> Decimal.cast(3.0)
       #Decimal<3.0>
 
-      iex> Decimal.from_any("3")
+      iex> Decimal.cast("3")
       #Decimal<3>
 
-      iex> Decimal.from_any("3.0")
+      iex> Decimal.cast("3.0")
       #Decimal<3.0>
 
-      iex> Decimal.new(3) |> Decimal.from_any()
+      iex> Decimal.new(3) |> Decimal.cast()
       #Decimal<3>
 
   """
-  @spec from_any(float | decimal) :: t
-  def from_any(float) when is_float(float), do: from_float(float)
-  def from_any(value), do: new(value)
+  @spec cast(float | decimal) :: t
```

Comment:
What do you think about `@spec cast(any) :: {:ok, t} | :error`?

Otherwise, I would document that if the value cannot be casted, a `Decimal.Error` error is raised.

Reply:
To me, either option sounds good. If we're going with `@spec cast(any) :: {:ok, t} | :error`, we'd likely add `@spec cast!(float | decimal) :: t` too but let's wait for other opinions.

Reply:
Thanks for the suggestions! 

In a vacuum, having `cast/1` return a tuple and `cast!/1` either return the value or raise an error feels like a good pattern. For consistency with the rest of the library, however, I slightly prefer instead documenting the raising of `Decimal.Error` when the value cannot be cast. The existing `new` would error under the same circumstances as `cast`, and it would be confusing to have different interfaces for handling the same class of issue.

I'll push a commit now adding that to the documentation and tests.